eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -284,65 +332,299 @@ function Invoke-Generate() {
+    Write-Host "specName:$specName, serviceType: $serviceType"
+
+    return $specName, $serviceType
+}
+
+function Invoke-GenerateAndBuildSDK () {
+    param(
+        [string]$readmeAbsolutePath,
+        [string]$sdkRootPath,
+        [string]$autorestConfigYaml = "",
+        [string]$downloadUrlPrefix = "",
+        [object]$generatedSDKPackages
+    )
+    $readmeFile = $readmeAbsolutePath -replace "\\", "/"
+    Write-Host "readmeFile:$readmeFile"
+    $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
+    Write-Host "service:$service, serviceType:$serviceType"
+    
+    if (!$readmeFile.StartsWith("http") -And !(Test-Path -Path $readmeFile)) {
+        Write-Error "readme file does not exist."
+        exit 1
+    }
+    
+    $packagesToGen = @()
+    $newpackageoutput = "newPackageOutput.json"
+    if ( $serviceType -eq "resource-manager" ) {
+        Write-Host "Generate resource-manager SDK client library."
+        $package = $service
+        if ($packageNameHash[$service] -ne "") {
+            $package = $packageNameHash[$service]
         }
-        if ($readmeFile -match $readmeFileRegex)
-        {
-            return $matches["specName"], $matches["serviceType"]
+        New-MgmtPackageFolder -service $service -packageName $package -sdkPath $sdkRootPath -commitid $commitid -readme $readmeFile -outputJsonFile $newpackageoutput
+        if ( !$?) {
+            Write-Error "Failed to create sdk project folder. exit code: $?"
+            exit 1
+        }
+        $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
+        $packagesToGen = $packagesToGen + @($newpackageoutputJson)
+        Remove-Item $newpackageoutput
+    } else {
+        Write-Host "Generate data-plane SDK client library."
+        $namespace = ""
+        if ( $autorestConfigYaml) {
+            # support single package
+            $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+            if ( $range.count -gt 1) {
+                $startNum = $range[0];
+                $lines = $range[1] - $range[0] - 1
+                $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
+            }
+            Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+            Import-Module powershell-yaml
+            $yml = ConvertFrom-YAML $autorestConfigYaml
+
+            $outputFolder = $yml["output-folder"]
+            if ($outputFolder) {
+                $directories = $outputFolder.Split("/");
+                $service = $directories[-2];
+                $namespace = $directories[-1];
+            }
+
+            New-DataPlanePackageFolder -service $service -namespace $namespace -sdkPath $sdkRootPath -readme $readmeFile -autorestConfigYaml "$autorestConfigYaml" -outputJsonFile $newpackageoutput
+            if ( !$? ) {
+                Write-Error "Failed to create sdk project folder. exit code: $?"
```

Comment:
Same comment as above re exit stopping execution in the inner function.

Reply:
resolved.